AffirmSDKDemo/AffirmSDKDemo/ViewController.m

```diff
@@ -21,31 +28,37 @@ @implementation ViewController
 - (void)viewDidLoad {
     [super viewDidLoad];
     [self setupView];
-    self.textField.accessibilityLabel = @"price field";
-    self.textField.delegate = self;
     
     CGRect frame = CGRectMake(0, 0, self.view.frame.size.width - 40, 40);
-    self.alaButton = [AffirmAsLowAsButton createButtonWithPromoID:@"promo_set_ios" presentingViewController:self frame:frame];
-    self.alaButton.center = CGPointMake(self.view.center.x, self.textField.frame.origin.y - 70);
+    self.alaButton = [AffirmAsLowAsButton createButtonWithPromoID:self.promoIDField.text presentingViewController:self frame:frame];
+    self.alaButton.center = CGPointMake(self.view.center.x, self.titleLabel.frame.origin.y - 80);
     [self.view addSubview:self.alaButton];
     
     [self reloadAffirmAsLowAs];
+    [self setAPIKey];
 }
 
 - (void)setupView {
-    UIToolbar *toolBar = [UIToolbar new];
-    UIBarButtonItem *flexibleItem = [[UIBarButtonItem alloc] initWithBarButtonSystemItem:UIBarButtonSystemItemFlexibleSpace target:nil action:nil];
-    UIBarButtonItem *doneItem = [[UIBarButtonItem alloc] initWithBarButtonSystemItem:UIBarButtonSystemItemDone target:self.textField action:@selector(resignFirstResponder)];
-    toolBar.items = @[ flexibleItem, doneItem ];
-    [toolBar sizeToFit];
-    self.textField.inputAccessoryView = toolBar;
+    [self configureTextField:self.amountField withLabel:@"price input" andTag:AMOUNT_FIELD_TAG];
+    [self configureTextField:self.promoIDField withLabel:@"promo ID input" andTag:PROMO_ID_FIELD_TAG];
+    [self configureTextField:self.publicAPIKeyField withLabel:@"public API key input" andTag:PUBLIC_API_KEY_FIELD_TAG];
 }
 
-- (void)textFieldDidBeginEditing:(UITextField *)textField {
-    self.textField.text = @"";
+- (BOOL)textFieldShouldReturn:(UITextField *)textField {
+    return [textField resignFirstResponder];
 }
 
 - (void)textFieldDidEndEditing:(UITextField *)textField {
+    switch (textField.tag) {
```

Comment:
FYI you don't need to use a tag like this. Can just check `if (textfield == self.amountField)` since the textfield that ended editing is a given parameter.